In [1]:
# Program to study how the new bus route for X8 is improving commute times?
from __future__  import print_function, division
import pylab as pl
import pandas as pd
import numpy as np
import os
import json

%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
# Downloading file from source
!curl -O https://raw.githubusercontent.com/fedhere/PUI2017_fb55/master/Lab3_fb55/times.txt


  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  1179  100  1179    0     0   3284      0 --:--:-- --:--:-- --:--:--  3284


In [3]:
data = pd.read_table("times.txt", header=None)


In [4]:
oldMean = 36
newMean = data.mean()
oldStd = 6
newStd = data.std()

# Problem Statement: How is the new bus route for X8 improving commute times?

## Alternate Hypothesis: The Average commute time for the new bus route X8 is lesser than the average commute time for the old commute route of X8.
    
   ### Ha: newMean - oldMean < 0
    
   ## Null Hypothesis: The Average Commute time for the new route is the same or longer than the old route.
    
   ### H0: newMean - oldMean >= 0
    
   ### Let us define the value of significance level α = 0.5 (5%)
    
   ### So the Confidence Level would be 1-α = 95%
    
   ### Z = \frac{\mu_{pop} - \mu_{sample}}{\sigma / \sqrt{N}}
   
   ### z = (oldMean - newMean)/(oldStd/np.sqrt(len(data)))
    
    

In [5]:
z = (oldMean - newMean)/(oldStd/np.sqrt(len(data)))
print (z)

0    2.556397
dtype: float64



### If the value of p(Ha)-p(H0) > α

### The Null Hypothesis can be rejected. In this case,


### p(Ha)-p(H0) = 2.556397 - 2 (For 2 Std, Deviations Considered) = 0.556397

### This value is greater than α

### Therefore,

### Since the value of Z>2 which is more than 2 Standard deviations above New Mean,

### The Null Hypothesis Can be rejected
